src/pal/src/init/pal.cpp

```diff
@@ -408,25 +408,27 @@ PAL_Initialize(
         // CreateInitialProcessAndThreadObjects took ownership of this memory.
         command_line = NULL;
 
-        //
-        // Tell the synchronization manager to start its worker thread
-        //
-
-        palError = CPalSynchMgrController::StartWorker(pThread);
-        if (NO_ERROR != palError)
+        if (argv != NULL)
```

Comment:
This is hard to reverse-engineer hack. We should make it more explicit and self-explanatory. E.g. add a new method to PAL that can be called before PAL_Initialize, to disable initialization of the certain PAL subsystems: 

void PAL_SetInitializationOptions(DWORD dwFlags);

#define PAL_INIT_NO_SYNCHRONIZATION  1
#define PAL_INIT_NO_SEH                              2